sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -439,6 +440,16 @@ class Analyzer(
             case s: Star => s.expand(child, resolver)
             case o => o :: Nil
           })
+        case p: Murmur3Hash if containsStar(p.children) =>
```

Comment:
Nope. It depends on the vendors. 

For example, in MS SQL Server: checksum support `*`: https://msdn.microsoft.com/en-us/library/ms189788.aspx

For example, for row-based compare: 
- get the signature of one row by `hash(*)` , which returns row-based checksums. 
- get the data of one row by `concat(*)`, which returns the whole row.